### PR TITLE
Refactor models

### PR DESCRIPTION
**Pull request by TheKaterTot**
We have classes now, and they're beautiful. 🍔 🔨  We took care of business.

### CODE REVIEW

File: lib/controllers/foods-controller.js [hunk 1]
```diff
@@ -1,36 +1,38 @@
...
-    })
 }
 
-module.exports = {
-  index: index,
-  create: create,
-  update: update,
-  destroy: destroy
-};
+module.exports = FoodsController;
```
Comment (cmacaulay) [inline lib/controllers/foods-controller.js:38]: 🎉 🎉 lovin' these classes, so readable.

File: lib/controllers/meals-controller.js [hunk 2]
```diff
@@ -1,15 +1,19 @@
...
+  }
+
+  index(req, res) {
+    let day = new Date(req.params.year, req.params.month, req.params.day);
+    let category = req.params.category;
 
-module.exports = {
-  index: index
+    this.meal.findByDate(day, category)
+    .then((data) => {
```
Comment (cmacaulay) [inline lib/controllers/meals-controller.js:13]: 💛 💛

File: lib/models/food.js [hunk 3]
```diff
@@ -2,82 +2,80 @@ const environment = process.env.NODE_ENV || 'development';
...
-    });
-  })
-}
-function find (id) {
-  return database.raw('SELECT * FROM foods WHERE id = ?', id)
-};
+        this.nameValidation(name, (err) => {
+          if (err) {
+            return reject(err);
+          }
```
Comment (cmacaulay) [inline lib/models/food.js:36]: 😮 🐺 🌔

File: lib/models/food.js [hunk 4]
```diff
@@ -2,82 +2,80 @@ const environment = process.env.NODE_ENV || 'development';
...
+      if (!food.calories && food.name) {
+        query = database.raw('UPDATE foods SET name = ?, created_at = ? WHERE id = ?',
+          [food.name, new Date, req.params.id])
+      } else if (food.calories && !food.name) {
+        query = database.raw('UPDATE foods SET calories = ?, created_at = ? WHERE id = ?',
+          [food.calories, new Date, req.params.id])
+      } else {
+        query = database.raw('UPDATE foods SET name = ?, calories = ?, created_at = ? WHERE id = ?',
+          [food.name, food.calories, new Date, req.params.id])
+      }
```
Comment (cmacaulay) [inline lib/models/food.js:66]: @case-eee can you think of a cleaner way to do this if/else block? curious!

File: lib/models/meal.js [hunk 5]
```diff
@@ -2,16 +2,18 @@ const environment = process.env.NODE_ENV || 'development';
...
+
+  findByDate (date, category) {
+    return database.raw(`SELECT meals.*, foods.name AS food_name, foods.calories AS calories,
+                                categories.name AS category_name
+                         FROM meals
+                         JOIN categories ON category_id = categories.id
+                         JOIN foods ON food_id = foods.id
+                         WHERE date = ? AND categories.name = ?`,
+      [date.toLocaleDateString(), category])
+  }
```
Comment (cmacaulay) [inline lib/models/meal.js:15]: 👏 👏

File: lib/routers/foods-router.js [hunk 6]
```diff
@@ -1,9 +1,21 @@
...
+  foodsController.create(req, res);
+})
+
+FoodsRouter.patch('/:id', (req, res) => {
+  foodsController.update(req, res);
+})
+
+FoodsRouter.delete('/:id', (req, res) => {
+  foodsController.destroy(req, res);
+})
```
Comment (cmacaulay) [inline lib/routers/foods-router.js:19]: 🐰 🐎 🐎

File: lib/routers/meals-router.js [hunk 7]
```diff
@@ -1,6 +1,10 @@
 const MealsRouter = require('express').Router();
 const MealsController = require('../controllers/meals-controller');
 
-MealsRouter.get('/:category/:year/:month/:day', MealsController.index);
+const mealsController = new MealsController;
+
+MealsRouter.get('/:category/:year/:month/:day', (req, res) => {
+  mealsController.index(req, res);
+})
```
Comment (cmacaulay) [inline lib/routers/meals-router.js:8]: 🥜

File: test/food-test.js [hunk 8]
```diff
@@ -15,7 +16,7 @@ describe('Food', () => {
   })
 
   it('is invalid without name', (done) => {
-    Food.create(null, 40)
+    food.create(null, 40)
       .catch( (err) => {
```
Comment (cmacaulay) [inline test/food-test.js:20]: :octocat: :octocat: :octocat:

File: test/server-test.js [hunk 9]
```diff
@@ -318,7 +319,7 @@ describe('Server', () => {
 
   describe('GET /api/meals/snacks/<date>', () => {
     beforeEach((done) => {
-      Food.create('popcorn', 10)
+      food.create('popcorn', 10)
```
Comment (cmacaulay) [inline test/server-test.js:322]: @TheKaterTot I'm hungry now. 🍔

**Review by cmacaulay — APPROVED** [review]
🍨 🍨 Love these classes!

**Comment by cmacaulay** [comment]
Looks great! 👯 #teamwork

File: lib/controllers/foods-controller.js (hunk 1 above)
Comment (case-eee) [reply lib/controllers/foods-controller.js:38]: 👍

File: lib/models/food.js (hunk 4 above)
Comment (case-eee) [reply lib/models/food.js:66]: @cmacaulay you could extract things a little more so it's more readable - you could have something like 

```js
if(updateNameOnly()) {
//...
} else if (updateCaloriesOnly()) {
//...
} else {
//...
}
```

you could also break down the SQL statements into smaller functions, but i think what you have is okay!

**Review by case-eee — COMMENTED** [review]
awesome work with this @TheKaterTot @cmacaulay - love that you're making this object oriented 😄

**Pull request merged**